src/libs/IntlPolyfill/polyfillDateTimeFormat.ts

```diff
@@ -0,0 +1,42 @@
+import type {DateTimeFormatConstructor} from '@formatjs/intl-datetimeformat';
+import DateUtils from '@libs/DateUtils';
+
+/* eslint-disable @typescript-eslint/naming-convention */
+const tzLinks: Record<string, string> = {
+    'Africa/Abidjan': 'Africa/Accra',
+    CET: 'Europe/Paris',
+    CST6CDT: 'America/Chicago',
+    EET: 'Europe/Sofia',
+    EST: 'America/Cancun',
+    EST5EDT: 'America/New_York',
+    'Etc/GMT': 'UTC',
+    'Etc/UTC': 'UTC',
+    Factory: 'UTC',
+    GMT: 'UTC',
+    HST: 'Pacific/Honolulu',
+    MET: 'Europe/Paris',
+    MST: 'America/Phoenix',
+    MST7MDT: 'America/Denver',
+    PST8PDT: 'America/Los_Angeles',
+    WET: 'Europe/Lisbon',
+};
+/* eslint-enable @typescript-eslint/naming-convention */
+
+export default function () {
+    // Because JS Engines do not expose default timezone, the polyfill cannot detect local timezone that a browser is in.
+    // We must manually do this by getting the local timezone before adding polyfill.
+    let currentTimezone = DateUtils.getCurrentTimezone().selected as string;
+    if (currentTimezone in tzLinks) {
+        currentTimezone = tzLinks[currentTimezone];
+    }
+
+    require('@formatjs/intl-datetimeformat/polyfill-force');
```

Comment:
@tienifr Can you address this comment: https://github.com/Expensify/App/issues/34035#issuecomment-2140560795